pkg/apiserver/authentication/identityprovider/ldap/ldap.go

```diff
@@ -147,21 +147,27 @@ func (l ldapProvider) Authenticate(username string, password string) (identitypr
 		return nil, err
 	}
 
-	if len(result.Entries) != 1 {
-		return nil, errors.NewUnauthorized("incorrect password")
+	var entry *ldap.Entry
+	switch n := len(result.Entries); n {
```

Comment:
Using the switch statement here is wired to me. What's wrong with a normal `if else` statement?

Reply:
Use `if else` statement instead.